src/main/java/io/jenkins/plugins/opentelemetry/job/OtelTraceService.java

```diff
@@ -233,9 +238,16 @@ public void putSpan(@NonNull Run run, @NonNull Span span) {
 
     public void putRunPhaseSpan(@NonNull Run run, @NonNull Span span) {
         run.addAction(new RunPhaseMonitoringAction(span));
+        setAttributesToSpan(span, run.getAction(OpenTelemetryAttributesAction.class));
         LOGGER.log(Level.FINEST, () -> "putRunPhaseSpan(" + run.getFullDisplayName() + "," + OtelUtils.toDebugString(span) + ")");
     }
 
+    public void putAgentSpan(@NonNull Run run, @NonNull Span span, @NonNull FlowNode flowNode) {
```

Comment:
Can we explain in the javadoc why we need a special handling for "agent allocation spans"

Reply:
To keep it consistent with the existing `putSpan` methods, I've added the explanation as a regular comment instead of a javadoc. Is that fine with you @cyrille-leclerc ?

Reply:
I also double-checked that `setAttributesToSpan(span, run.getAction(OpenTelemetryAttributesAction.class));` is indeed necessary for phase and agent spans. With that removed, some of the new tests in `WithSpanAttributesStepTest` would fail.